### DISCUSSION TITLE
Handle multiple time periods of 1 model setup

### DISCUSSION THREAD

**Discussion by Hendrik1987** [DHI/modelskill#468] [Q&A] [answered]
Dear modelskill team.

I have one point observation and one model setup, run for two seperate periods in time, say a and b. I have scattered measurements, and don't want to run a long model during calibration phase.

Here is what I tried:
```
obs1 = ms.PointObservation(obs1.dfs0, item=0)
mr1a = ms.model_result(mdl1_part1.dfs0, item=0)
mr1b = ms.model_result(mdl1_part2.dfs0, item=0)
```

I can of course do seperate comparers and plotting:
```
cmp1 = ms.match(obs1, mr1a)
cmp2 = ms.match(obs1, mr1b)
cmp1.plot.scatter()
cmp2.plot.scatter()
```

But I actually want to concat mr1a and mr1b somehow. This does not work:
```
cmp = ms.match(obs1,[mr1a, mr1b])
cmp.plot.scatter()
```
I get ValueError: No data found in selection.

I did not expect this to work, as I suppose the logic is reserved for having two different model setups rather than periods.

Should I concat the model data as preprocessing step or is there a smarter way?

I guess, this also relates to the "max_model_gap" discussion that we had back then. But I can't figure out how this is handled in the latest modelskill version.

Thanks for help and hints!

**Comment by ecomodeller** [comment, marked as answer]
Have you tried to "add" two comparers?

If the model have the same *name* I think it should work as expected.
```python
cmp = cmp1 + cmp2
```
https://github.com/DHI/modelskill/blob/main/notebooks/Combine_comparers.ipynb

**Reply by Hendrik1987** [reply]
Yes, that works, thanks!

Only the `cmp.plot_timeseries()` behaves (maybe) little unexpecte, as it zooms into the first model period only.

Inspecting `cmp.data` and the  `cmp.plot()`, however, confirms that all data is correctly available in `cmp`.

**Reply by ecomodeller** [reply]
Thanks for pointing that out, seems like something someone😉 should look into.

**Reply by ecomodeller** [reply]
@Hendrik1987 I have deprecated the `+` operator, since it was a bit too magic. How about concatenating the dfs0 files using MIKE IO, should work right?
